.github/workflows/ci.yml

```diff
@@ -0,0 +1,322 @@
+name: LinuxKit CI
+on: [push, pull_request]
+
+jobs:
+  build:
+    name: Build & Test
+    strategy:
+      matrix:
+        arch:
+          - amd64-linux
+          - arm64-linux
+          - s390x-linux
+          - amd64-darwin
+          - amd64-windows.exe
+
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Set up Go 1.11
+      uses: actions/setup-go@v1
+      with:
+        go-version: 1.11
+      id: go
+
+    - name: Check out code
+      uses: actions/checkout@v1
+      with:
+        path: ./src/github.com/linuxkit/linuxkit
+
+    - name: Get pre-requisites
+      run: |
+            echo "::set-env name=PATH::$PATH:$(go env GOPATH)/bin"
+            go get -u golang.org/x/lint/golint
+            go get -u github.com/gordonklaus/ineffassign
+      env:
+        GOPATH: ${{runner.workspace}}
+
+    - name: Lint
+      run: |
+        make local-check
+      env:
+        GOPATH: ${{runner.workspace}}
+
+    - name: Build
+      run: |
+        make LOCAL_TARGET=bin/linuxkit-${{matrix.arch}} local-build
+      env:
+        GOPATH: ${{runner.workspace}}
+
+    - name: Checksum
+      run: cd bin && sha256sum linuxkit-${{matrix.arch}} > linuxkit-${{matrix.arch}}.SHA256SUM
+
+    - name: Test
+      run: make local-test
+      env:
+        GOPATH: ${{runner.workspace}}
+
+    - name: Cache binary
+      uses: actions/cache@v1
+      with:
+        path: bin
+        key: linuxkit-${{matrix.arch}}-${{hashFiles('src/**')}}
+
+    - name: Upload binary
+      uses: actions/upload-artifact@v1.0.0
+      with:
+        name: linuxkit-${{matrix.arch}}
+        path: bin
+
+  build_packages:
+    name: Build Packages
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check out code
+      uses: actions/checkout@v1
+      with:
+        path: ./src/github.com/linuxkit/linuxkit
+
+    - name: Restore LinuxKit From Cache
+      uses: actions/cache@v1
+      with:
+        path: lkt
+        key: linuxkit-amd64-linux-${{hashFiles('src/**')}}
+
+    - name: Symlink Linuxkit
+      run: |
+        sudo ln -s `pwd`/lkt/linuxkit-amd64-linux /bin/linuxkit
+
+    - name: Build Packages
+      run: |
+        make -C pkg build
```

Comment:
does ubuntu-latest have docker pre-installed? If so which version? Is there a way to pin the version?

Reply:
According to [Github](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/software-installed-on-github-hosted-runners) we have Docker version 3.0.8 🤔 
Sadly there is no way I know of to pin this.
I'd like very much to remove the Docker dependency anyway